.github/workflows/ci-azure.yml

```diff
@@ -0,0 +1,124 @@
+name: CI Azure
+on:
+  workflow_call:
+  workflow_dispatch:
+    inputs:
+      ci_backend:
+        description: 'Name of the job backend/target'
+        default: ''
+        required: false
+        type: string
+      ci_option:
+        description: 'Name of the job option for display'
+        default: ''
+        required: false
+        type: string
+      bootstrap_args:
+        default: ''
+        required: false
+        type: string
```

Comment:
```suggestion
```

This workflow is currently focused on Azure and these options can be hardcoded. We can later generalize it into a "real cloud integration test" workflow and add a matrix with the various cloud providers and authentication scenarios we support.